Firestore/Source/Remote/FSTRemoteStore.mm

```diff
@@ -107,7 +107,9 @@ @interface FSTRemoteStore () <FSTWatchStreamDelegate, FSTWriteStreamDelegate>
 @end
 
 @implementation FSTRemoteStore {
+  std::unique_ptr<WatchChangeAggregator> _watchChangeAggregator;
```

Comment:
It would be simple to add a default constructor to `WatchChangeAggregator` and store it as a value. The reason I'm leaning towards a pointer is that, should there be a bug when the aggregator is accessed after it's destroyed but before it's recreated, app will crash due to null pointer access; if instead it's a value type, it would "work" (at least without adding some notion of being "valid" to the aggregator, which other platforms don't have). It's not a very strong reason, though; let me know what you think.

Reply:
either way is fine for me.

Reply:
Note that objc allows you to send messages to nil. So this port technically changes the behaviour, i.e. this might work under objc but would null pointer under c++:

```
[remoteStore cleanupWatchStreamState]
[remoteStore sendUnwatchRequestForTargetId targetId:x]
```

But I suspect that we only call cleanup() during, well, cleanup of this object.

My preference would be to add asserts so that we get a better error message... but it would admittedly be pretty low value. If we leave out the asserts, then I like using a pointer over an invalid value type for the reasons you state. (And even if we *do* add asserts, then I'd probably base them off the pointer value anyways, rather than adding another bool or similar. So either way, I prefer the pointer approach.)

Reply:
AFAIU, iOS is normally the biggest outlier, and in this case, Objective-C indeed would not crash, but Android and Web will. For this reason, I see this as bringing iOS more in line with other platforms.